File: archive/sort-array-by-parity.js
const sortArrayByParity = function(nums) {
  let a1 = []
  let a2 = []

  nums.forEach(n => {
    if (n % 2 === 0) {
      a1.push(n)
    } else {
      a2.push(n)
    }
  })

  return [...a1, ...a2]
};

File: sort-array-by-parity.go
func sortArrayByParity(nums []int) []int {
    odd:=[]int{}; noodd:=[]int{}

    for _, v:= range nums {
        if v % 2 == 0 {
            odd = append(odd, v)
        } else {
            noodd = append(noodd, v)
        }
    }

    return append(odd, noodd...)
}